OpenRA.Mods.Common/Traits/World/EditorActorPreview.cs

```diff
@@ -138,6 +140,25 @@ public void RemovedFromEditor()
 				kv.Key.RemovedFromEditor(this, worldRenderer.World, kv.Value);
 		}
 
+		public void ReplaceActor(string type)
```

Comment:
This is really not a good idea!

The map editor is designed around the assumption that the position and footprint/bounds are immutable. The main usecase for building variants is to have a different footprints for different rotations. This will break things.

Reply:
What do you suggest instead?

Reply:
There is no easy or sensible way to implement #18637 before a major editor overhaul to support moving actors.

Reply:
I decoupled this from the in-game feature now and added some examples.

Reply:
The fundamental problem I described above is related to removing the readonly attribute from `Info` and `reference`. Adding a new `EditorVariants` doesn't change anything in that regard.

Reply:
Can we somehow cheat here by removing and adding back to the editor?

Reply:
I already cheated by only applying this onto actors with the same footprint which is what https://github.com/OpenRA/OpenRA/issues/18637 actually only asks for.